### PR TITLE
Update WalletConnect provider

### PR DESCRIPTION
**Pull request by Velenir**
Updates WC provider, with quite a leap in version
Therefore needs thorough testing first

Otherwise some dependencies get deduped and pinned to incompatible versions, and web3@1.2.8 breaks:
```
@ethersproject/bignumber version is wrong

> npm ls @ethersproject/bignumber
@gnosis.pm/dex-react@0.11.1 /home/velenir/Projects/Gnosis/dex-react
├─┬ @walletconnect/web3-provider@1.0.0-beta.47
│ └─┬ @walletconnect/browser@1.0.0-beta.47
│   └─┬ @walletconnect/utils@1.0.0-beta.47
│     ├─┬ @ethersproject/address@5.0.0-beta.134
│     │ └── @ethersproject/bignumber@5.0.0-beta.135 
│     └─┬ @ethersproject/strings@5.0.0-beta.136
│       └─┬ @ethersproject/constants@5.0.0-beta.133
│         └── @ethersproject/bignumber@5.0.0-beta.135  deduped
└─┬ web3@1.2.8
  └─┬ web3-eth@1.2.8
    └─┬ web3-eth-abi@1.2.8
      └─┬ @ethersproject/abi@5.0.0-beta.153
        ├─┬ @ethersproject/address@5.0.0-beta.135
        │ └── @ethersproject/bignumber@5.0.0-beta.139 
        ├── @ethersproject/bignumber@5.0.0-beta.135  deduped
        └─┬ @ethersproject/strings@5.0.0-beta.137
          └─┬ @ethersproject/constants@5.0.0-beta.134
            └── @ethersproject/bignumber@5.0.0-beta.139 

normally it would be beta.139 but because of walletconnect dependency where it's pinned to beta.135 it got deduped like that
```

### CODE REVIEW

**Comment by anxolin** [comment]
Travis failed, TS unhappynes? https://travis-ci.org/github/gnosis/dex-react/builds/690006009#L779

I'll rebase this branch also with the latest develop and force a recompile, it's outdated

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr1039--dexreact.review.gnosisdev.com

**Comment by Velenir** [comment]
Latest **WalletConnect** provider just doesn't work with our setup.
Need to update `dapp-ui` with new `web3modal(web3connect)`
Or simply throw `dapp-ui` away and do everything here.

**Comment by anxolin** [comment]
> Latest WalletConnect provider just doesn't work with our setup. Need to update dapp-ui with new web3modal(web3connect). Or simply throw dapp-ui away and do everything here.

For me is fine to throw `dapp-ui` if it make it simpler

**Comment by Velenir** [comment]
> So, I wait for testing this, right?

Yes

**Comment by Velenir** [comment]
Can test now

File: src/api/wallet/WalletApi.ts [hunk 1]
```diff
@@ -172,6 +287,8 @@ const subscribeToBlockchainUpdate = ({
 
 // const AUTOCONNECT = process.env.AUTOCONNECT === 'true'
 
+type WalletConnectInits = IProviderOptions['walletconnect']
+
 const wcOptions: Omit<WalletConnectInits, 'package'> = {
   options: {
     // TODO get infuraId from .env
```
Comment (alfetopito) [inline src/api/wallet/WalletApi.ts:294]: You can remove this comment

```suggestion
```

**Comment by Velenir** [comment]
> Can't connect with WC. When I click on the button that usually displays the QR code, nothing happens and an error is logged on the console:
> 
> screenshot_2020-06-03_09-44-00
> 
> As always, using FF.

oops, Travis didn't even build, that's why you saw an older version
Was not up to date with develop
Now should be good.

I think it's really great we have at least one developer primarily on FF, otherwise we would just forget about other browsers :disappointed_relieved:

**Comment by Velenir** [comment]
Check #1085 first, please

**Comment by Velenir** [comment]
> I'm afraid I get a random error I don't see in dev

Why does it only break in prod, smh
Should be ok now

File: package.json [hunk 2]
```diff
@@ -28,15 +28,14 @@
     "@fortawesome/free-regular-svg-icons": "^5.12.0",
     "@fortawesome/free-solid-svg-icons": "^5.12.0",
     "@fortawesome/react-fontawesome": "^0.1.8",
-    "@gnosis.pm/dapp-ui": "^0.5.3",
```
Comment (anxolin) [inline package.json:31]: bye bye 👋

Comment (W3stside) [reply package.json:31]: hate to see it go

**Pull request merged**